bridges/opentracing/lib/opentelemetry/bridge/opentracing/tracer.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Bridge
+    module OpenTracing
+      # Tracer provides a means of referencing
+      # an OpenTelemetry::Tracer as an OpenTracing::Tracer
+      class Tracer
+        attr_reader :tracer
+
+        def initialize(tracer)
+          @tracer = tracer
+        end
+
+        def active_span
+          scope = ScopeManager.instance.active
+          scope&.span
+        end
+
+        def start_active_span(operation_name,
+                              child_of: nil,
+                              references: nil,
+                              start_time: Time.now,
+                              tags: nil,
+                              ignore_active_scope: false,
+                              finish_on_close: true,
+                              &block)
+          child_of = child_of.span unless child_of.instance_of? OpenTelemetry::Trace::Span
+          span = @tracer.start_span(operation_name,
+                                    with_parent: child_of,
+                                    attributes: tags,
+                                    links: references,
+                                    start_timestamp: start_time)
+          wrapped = Span.new(span, dist_context: span.context)
+          scope = Scope.new(ScopeManager.instance, wrapped, finish_on_close)
```

Comment:
Scope activation / deactivation (closing) needs to correspond with an OpenTelemetry::Context containing the active span being attached and detached. Attach makes that the current context, whereas detach restores the previous context. Manual Context#attach and Context#detach are somewhat dangerous, as each call to attach and detach need to match, however, I think the scope concept from OT should make this work out.